### PR TITLE
[Plugin] Add custom widget and other improvements

### PR DESCRIPTION
**Pull request by IntelOrca**
* Add API for getting network traffic statistics.
* Add API for creating custom widgets.
* Add API for drawing graphics for custom widgets.
* Add click event to spinners and allow them to be held down.

image
image

### CODE REVIEW

File: distribution/openrct2.d.ts [hunk 1]
```diff
@@ -2125,8 +2125,16 @@ declare global {
     interface SpinnerWidget extends WidgetBase {
         type: 'spinner';
         text?: string;
+
+        /**
+         * If true, the user can click and hold the decrement and increment
+         * button for repeated calls.
+         */
+        isHoldable?: boolean;
```
Comment (Gymnasiast) [inline distribution/openrct2.d.ts:2133]: Why would we want to turn this off? Native OS spinners/steppers always allow holding down, AFAICT. The fact that OpenRCT2 doesn't is a misfeature IMO.

Comment (IntelOrca) [reply distribution/openrct2.d.ts:2133]: You mentioned this in https://github.com/OpenRCT2/OpenRCT2/issues/14095#issuecomment-779663385
I was thinking about making the default, but if you think the option should just be removed entirely, I can do that.

Comment (Gymnasiast) [reply distribution/openrct2.d.ts:2133]: I think removing it entirely and always allowing holding them down is indeed best. It improves usability and consistency (including with native OS widgets).

Comment (IntelOrca) [reply distribution/openrct2.d.ts]: ok, done

File: src/openrct2-ui/interface/Window.cpp [hunk 2]
```diff
@@ -681,6 +688,21 @@ void InvalidateAllWindowsAfterInput()
     });
 }
 
+bool Window::IsLegacy()
```
Comment (Gymnasiast) [inline src/openrct2-ui/interface/Window.cpp:691]: What does "legacy" mean in this context?

Comment (duncanspumpkin) [reply src/openrct2-ui/interface/Window.cpp:691]: If it has been converted to a class or if it's the old style rct_window

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**